### PR TITLE
[OYPD-110] Migrate demo content while installation

### PR DESCRIPTION
**Pull request by danylevskyi**
Task: OYPD-110

### How to review

1. Install site via `reinstall.sh` and you should see `blog` and `branch` demo content.
2. Install site via browser and you should see additional installation step `Import demo content`. Select appropriate content to import.

**NOTE**, `landing` and `programs` content are not migrated (there is no migrations).

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>2</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build297/build_reports/Drupalsniff.txt
Build site installed at http://openy-dev.ffwua.com/build297

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/297/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build297/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build298

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/298/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build298/profiles/openy/build/reports/behat

File: src/ContentImporter.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    // Service container are not available on the first steps of profile installation.
+    $this->migrationManager = \Drupal::service('plugin.manager.config_entity_migration');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getMap() {
+    // @todo Move the map into config.
+    // @todo Set dependencies for "landing" and "programs".
```
Comment (alexschedrov) [inline src/ContentImporter.php]: Landing pages, programs and categories have been added in PRs #89 and #91. Let's include them here.

File: src/ContentImporter.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      'landing' => [],
+      'branches' => [
+        'openy_demo_node_branch'
+      ],
+      'blog' => [
+        'openy_demo_node_blog',
+      ],
+      'programs' => [],
+    ];
+  }
```
Comment (alexschedrov) [inline src/ContentImporter.php]: Should we also trigger there taxonomy terms or they will be triggered automatically?

File: src/ContentImporter.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    $this->map = $this->getMap();
+
+    // Service container are not available on the first steps of profile installation.
+    $this->migrationManager = \Drupal::service('plugin.manager.config_entity_migration');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getMap() {
```
Comment (alexschedrov) [inline src/ContentImporter.php:43]: We should also trigger migration for footer blocks https://github.com/ymcatwincities/openy/tree/8.x-1.x/modules/openy_features/openy_demo_content/modules/openy_demo_bfooter, but there's dependency on block configuration. So, could be tricky to resolve dependencies.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by alexschedrov** [comment]
@danylevskyi it looks awesome!!! 🏆

File: src/ContentImporter.php (hunk 2 above)
Comment (danylevskyi) [reply src/ContentImporter.php]: All dependencies are handled automatically (if they specified in migrations)

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>4</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build315/build_reports/Drupalsniff.txt
Build site installed at http://openy-dev.ffwua.com/build315

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/315/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build315/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>4</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build316/build_reports/Drupalsniff.txt
Build site installed at http://openy-dev.ffwua.com/build316

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/316/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build316/profiles/openy/build/reports/behat

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build318

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/318/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build318/profiles/openy/build/reports/behat

**Review by alexschedrov — APPROVED** [review]

**Comment by alexschedrov** [comment]
@danylevskyi For some reason facility type vocabulary is empty.
image

**Pull request merged**